tests/dataset_histograms/computing_histograms_test.py

```diff
@@ -563,28 +570,164 @@ def test_compute_partitions_privacy_id_count_histogram(
+                input=lambda: [(
+                    (0, 0), 1.0)] * 100,  # ((privacy_id, partition), value)
+                expected=lambda: [
+                    hist.FrequencyBin(
+                        lower=100.0, upper=100.0, count=1, sum=100.0, max=100.0
+                    ),
+                ]),
+            dict(
+                testcase_name=
+                '1 privacy id many equal contributions to many partitions',
+                input=lambda: [((0, i), 1.0) for i in range(1234)],
+                # ((privacy_id, partition), value)
+                expected=lambda: [
+                    hist.FrequencyBin(
+                        lower=1.0, upper=1.0, count=1234, sum=1234.0, max=1),
+                ]),
+            dict(
+                testcase_name=
+                '1 privacy id many different contributions to many partitions',
+                input=lambda: [((0, i), i) for i in range(10001)],
+                # ((privacy_id, partition), value)
+                # step is 1e4 / 1e4 = 1, therefore 1e4 - 1 = 9999.
+                expected=lambda: [
+                    hist.FrequencyBin(lower=float(i),
+                                      upper=float(i + 1),
+                                      count=1,
+                                      sum=i,
+                                      max=i) for i in range(9999)
+                ] + [
+                    hist.FrequencyBin(
+                        lower=9999, upper=1000, count=2, sum=19999, max=10000)
+                ]),
+            dict(
+                testcase_name=
+                '2 privacy ids, same partitions equally contributed',
+                input=lambda: [((0, i), 1.0) for i in range(15)] + [(
+                    (1, i), 1.0) for i in range(10, 25)],
+                # ((privacy_id, partition), value)
+                expected=lambda: [
+                    hist.FrequencyBin(
+                        lower=1.0, upper=1.0001, count=20, sum=20.0, max=1.0),
+                    hist.FrequencyBin(
+                        lower=1.9999,
+                        upper=2.0, count=5, sum=10.0, max=2.0)
+                ]),
+            dict(
+                testcase_name='2 privacy ids, same partitions differently '
+                'contributed',
+                input=lambda: [((0, i), -1.0) for i in range(15)] + [(
+                    (1, i), 1.0) for i in range(10, 25)],
+                # ((privacy_id, partition), value)
+                # step = (1 - (-1)) / 1e4 = 0.0002,
+                # therefore last lower is 1 - 0.0002 = 0.9998.
+                expected=lambda: [
+                    hist.FrequencyBin(lower=-1.0,
+                                      upper=-0.9998,
+                                      count=10,
+                                      sum=-10.0,
+                                      max=-1.0),
+                    hist.FrequencyBin(lower=0.0,
+                                      upper=0.00019999999999997797,
```

Comment:
hm, so this value we have here due to imprecise math, should then try to assert with an interval around 0.0002? But I guess then assertion will be more complicated because we won't be able to assert just like assert(actual == expected).